src/wallet/wallet2.cpp

```diff
@@ -8426,31 +8426,66 @@ wallet2::request_stake_unlock_result wallet2::can_request_stake_unlock(const cry
   return result;
 }
 
-static bool prepare_tx_extra_loki_name_system_values(cryptonote::network_type nettype,
-                                                     lns::mapping_type type,
-                                                     uint32_t priority,
-                                                     std::string const &name,
-                                                     std::string const &value,
-                                                     wallet2 const &wallet,
-                                                     crypto::hash &prev_txid,
-                                                     lns::lns_value &value_blob,
-                                                     std::string *reason)
-{
+struct lns_prepared_args
+{
+  bool prepared;
+  operator bool() const { return prepared; }
+  lns::lns_value             value;
+  crypto::generic_public_key owner;
+  crypto::generic_public_key backup_owner;
+  crypto::generic_signature  signature;
+  crypto::hash               prev_txid;
+};
+
+static lns_prepared_args prepare_tx_extra_loki_name_system_values(wallet2 const &wallet,
+                                                                  lns::mapping_type type,
+                                                                  uint32_t priority,
+                                                                  std::string const &name,
+                                                                  std::string const *value,
+                                                                  std::string const *owner,
+                                                                  std::string const *backup_owner,
+                                                                  bool make_signature,
+                                                                  std::string *reason)
+{
+  lns_prepared_args result = {};
   if (priority == tools::tx_priority_blink)
   {
     if (reason) *reason = "Can not request a blink TX for Loki Name Service transactions";
-    return false;
+    return result;
   }
 
   if (!lns::validate_lns_name(type, name, reason))
-    return false;
+    return result;
 
-  if (!lns::validate_lns_value(nettype, type, value, &value_blob, reason))
-    return false;
+  if (value)
+  {
+    if (!lns::validate_lns_value(wallet.nettype(), type, *value, &result.value, reason))
+      return result;
+  }
+
+  if (owner)
+  {
+    if (!epee::string_tools::hex_to_pod(*owner, result.owner))
```

Comment:
It would be useful both here (and in backup owner) if I could specify a full Loki wallet address and have it extract the spend pubkey from it.

Reply:
Noted. This is client side so if it doesn't make it in before testnet, I can PR afterwards.